packages/api-contract/src/Abi/toV5.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright 2017-2024 @polkadot/api-contract authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ContractMetadataV4, ContractMetadataV5 } from '@polkadot/types/interfaces';
+import type { Registry } from '@polkadot/types/types';
+
+import { objectSpread } from '@polkadot/util';
+
+export function v4ToV5 (registry: Registry, v4: ContractMetadataV4): ContractMetadataV5 {
+  return registry.createType('ContractMetadataV5', objectSpread({}, v4, {
+    spec: objectSpread({}, v4.spec, {
+      events: v4.spec.events.map((e) =>
+        registry.createType('ContractEventSpecV3', objectSpread({
+          module_path: undefined,
+          signature_topic: undefined
```

Comment:
Or do we consider this a breaking change and don't allow converting from v4 to v5?

Reply:
I'm not sure about the implications, but here's how we describe the change:

>Previously the order of the events in the `events` array was significant (i.e. the first one had an implied index of `0`), and this index could be used to determine which event to decode. Now that is replaced by the `signature_topic`, and the order of the events in the metadata no longer has any significance.

Reply:
> However, this time there is newly added data we can't create for older version of the metadata.

I am getting myself familiar with the correlation between Ink v5 and the conversion process above, and found this on the following `signatureTopic`:

```rust
/// # The Signature Topic
///
/// By default, the [`ink::Event::SIGNATURE_TOPIC`] is calculated as follows:
///
/// `blake2b("EventStructName(field1_type_name,field2_type_name)")`
/// The hashing of the topic is done at codegen time in the derive macro, and as such only has
/// access to the **names** of the field types as they appear in the code. As such, if the
/// name of a field of a struct changes, the signature topic will change too, even if the
/// concrete type itself has not changed. This can happen with type aliases, generics, or a
/// change in the use of a `path::to::Type` qualification.
```

1. Would it be invalid to construct the signatureTopic for a v4 event when converting to v5?

2. In this case do we not have the valid info/data to construct it if the conversion can be considered valid?

Reply:
Not sure if relevant here, but the topic calculation changed in general from v4 to v5, so it's a breaking change for any clients that used topics to filter for events. 

We have a migration guide for the events changes [here](https://use.ink/5.x/faq/migrating-from-ink-4-to-5/#events-20).